vtex-apps/search-result
# Fix pagination & Fetch previous button

#### What is the purpose of this pull request?

Make the URL change when you load more results of a search result. With this feature, it is now possible to share a certain page and when using the `go back` browser feature you now go back to the last page you were in.
There is now a new button `show previous` that shows the results of previous pages if they are available.
This PR also fixes some issues with the previous search result pagination, such as showing the load more button even if the user was already on the last page.

#### How should this be manually tested?

Go [here](https://iaronaraujo--invictastores.myvtex.com/Collections/Aviator), click on the load more button and check that the URL changed. Click on a product to go to its page, then click on the `back` button of your browser and check that you returned to page 2 and that the products of the first page are not being shown. Then, click on the load more button again and observe that the page changed again and that the items are being shown correctly.

Go [here](https://iaronaraujo--invictastores.myvtex.com/Collections/Aviator?page=4) and check that since this is the last page of this search, the load more button at the end of the results does not appear, but there is a button at the top of the results that loads the previous results.

Go [here](https://iaronaraujo--invictastores.myvtex.com/Collections/Aviator?page=3) and check that the `load more` button appears on the top and on the bottom of the search result since there this isn't the first nor the last page. Click on the `load more` button on the top and check that the URL changed to page 2. Now click on the button on the bottom and check that the URL changed to page 4.

You can do the same tests above with a [store that uses infinite scroll](https://iaronaraujo--alssports.myvtex.com/clothing/shorts) and with a [store that uses flexible search result](https://iaronaraujo--storecomponents.myvtex.com/top)

Notice that if you change any search parameter, such as ordination type and filters, the page is reset.

Needs https://github.com/vtex-apps/render-runtime/pull/421 to work properly

#### Types of changes
